### ISSUE TITLE
nlp:date first/second/third/fourth  quarter resolved into 01-Jul-19/01-Oct-19

### ISSUE THREAD

**Issue by ermadan** [vic64/nlpcraft#37]
v 4.0.0
````
        solver.addIntent(NCIntentSolver.CONV_INTENT(
                "salesClient",
                NCIntentSolver.TERM(NCIntentSolver.AND("id == nlp:function", "~FUNCTION_TYPE == LIMIT"), 0, 1),
                NCIntentSolver.TERM("id == aspen:salesclient", 0, 1),
                NCIntentSolver.TERM("id == aspen:period", 0, 1),  //optional period token
                NCIntentSolver.TERM("id == nlp:geo", 0, 1),
                NCIntentSolver.TERM("id == nlp:date", 0, 1)          // Term #2 (index 1): optional date.
        ), this::onIntentMatch)
````
query "top 5 sales in first quarter", dates resolved into "dateFrom":"01-Jul-19","dateTo":"01-Oct-19"
logs shows that "first" is not used in conjunction with "quarter":
````
| #3      | salesClient | INTENT: 'salesClient'                                                             |
|         |             |   Group #0                                                                        |
|         |             |     #0: Token [id=nlp:function, text=top 5, group=nlp:function, value=null]       |
|         |             |   Group #1                                                                        |
|         |             |     #0: Token [id=aspen:salesclient, text=sales, group=null, value=null]          |
|         |             |   Group #2                                                                        |
|         |             |     <empty>                                                                       |
|         |             |   Group #3                                                                        |
|         |             |     #0: (conv) Token [id=nlp:geo, text=europe, group=nlp:geo, value=null]         |
|         |             |   Group #4                                                                        |
|         |             |     #0: (conv) Token [id=nlp:date, text=quarter, group=nlp:date, value=null]      |
````

**Comment by skhdl** [comment]
Seems we have to add support for following dates
 - first  quarter
 - 1  quarter
 - 1st  quarter
etc
for current year.

Now it works only if the year is defined directly "first  quarter 2019"

**Comment by skhdl** [comment]
Can be done together with GH-30

**Comment by skhdl** [comment]
ready for review in GH-30
(dates model is not re-generated)

**Comment by vic64** [comment]
Merged to master as part of #30 fix.
